src/Microsoft.Diagnostics.Runtime/src/DacInterface/MetadataImport.cs

```diff
@@ -171,15 +159,16 @@ private void CloseEnum(IntPtr handle)
 
         public IEnumerable<int> EnumerateGenericParams(int token)
         {
-            InitDelegate(ref _enumGenericParams, VTable.EnumGenericParams);
-
             IntPtr handle = IntPtr.Zero;
             int[] tokens = ArrayPool<int>.Shared.Rent(32);
             try
             {
-                while (_enumGenericParams(Self, ref handle, token, tokens, tokens.Length, out int count) && count > 0)
+                while (Enum(ref handle, token, tokens, tokens.Length, out int count) && count > 0)
                     for (int i = 0; i < count; i++)
                         yield return tokens[i];
+
+                bool Enum(ref IntPtr handle, int token, int[] tokens, int length, out int count) =>
```

Comment:
Do we need this?  I can't figure out why this doesn't just call `VTable.EnumGenericParams` directly

Reply:
> CS1629 Unsafe code may not appear in iterators